lib/project-folder-dnd-handler.coffee

```diff
@@ -0,0 +1,178 @@
+BrowserWindow = null # Defer require until actually used
+RendererIpc = require 'ipc'
+
+{$, View} = require 'atom-space-pen-views'
+_ = require 'underscore-plus'
+
+module.exports =
+class ProjectFolderDragAndDropHandler
+  constructor: (@treeView) ->
+    @treeView.on 'dragend', '.project-root-header', @onDragEnd
+
+    RendererIpc.on('tree-view:project-folder-dropped', @onDropOnOtherWindow)
+
+  # unused
+  unsubscribe: ->
+    RendererIpc.removeListener('tree-view:project-folder-dropped', @onDropOnOtherWindow)
+
+  onDragStart: (event) =>
+    unless $(event.target).closest('.project-root-header').size()
+      return
+
+    event.originalEvent.dataTransfer.setData 'atom-event', 'true'
+    projectRoot = $(event.target).closest('.project-root')
+    directory = projectRoot[0].directory
+
+    event.originalEvent.dataTransfer.setData 'project-root-index', projectRoot.index()
+
+    rootIndex = -1
+    _.find(@treeView.roots, (root, index) -> root.directory is directory and ((rootIndex = index) or true))
```

Comment:
`(rootIndex = index) or true)` I might be missing something, but won't that always evaluate to true?